sympy/tensor/array/expressions/conv_matrix_to_array.py

```diff
@@ -1,11 +1,11 @@
-from sympy import Mul, Basic, MatMul, MatAdd, Transpose, Trace, Pow, \
+from sympy import Integer, Mul, Basic, MatMul, MatAdd, Transpose, Trace, Pow, \
     MatPow, symbols, Dummy, Lambda, HadamardProduct, HadamardPower, S
 from sympy.matrices.expressions.matexpr import MatrixExpr
 from sympy.tensor.array.expressions.array_expressions import ArrayDiagonal, ArrayTensorProduct, \
     PermuteDims, ArrayAdd, ArrayContraction, ArrayElementwiseApplyFunc
 
 
-def convert_matrix_to_array(expr: MatrixExpr) -> Basic:
+def convert_matrix_to_array(expr: Basic) -> Basic:
```

Comment:
Why is Expr not good enough here?

Reply:
The function calls itself recursively with the args of its argument. In general the args of `Expr` are not known to be `Expr` e.g.:
```python
In [11]: M = MatrixSymbol('X', 2, 2)

In [12]: isinstance(M, Expr)
Out[12]: True

In [13]: isinstance(M.args[0], Expr)
Out[13]: False
```
It should be possible to add type hints to each of the different classes here like `MatMul` could have
```python
    args: Tuple[Expr, ...]
```
I expect that adding those type hints will throw up more errors though when I'm actually trying to silence the errors first so that we can run mypy and then add type-hints properly later.